### PR TITLE
getitem: define is_singular

### PR DESCRIPTION
**Pull request by ben-bou**

### CODE REVIEW

**Comment by mtar** [comment]
GPU cluster tests are currently disabled on this Pull Request.

File: heat/core/dndarray.py [hunk 1]
```diff
@@ -730,6 +730,32 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
...
+            except Exception:
+                pass
+            if hasattr(obj, "__len__") and len(obj) == 1:
+                return True
+            if hasattr(obj, "shape") and len(obj.shape) == 0:
+                return True
+            if hasattr(obj, "size"):
+                return obj.size == 1
+            if hasattr(obj, "numel"):
+                return obj.numel == 1
```
Comment (ClaudiaComito) [inline heat/core/dndarray.py:750]: this needs to be `obs.numel()`

File: heat/core/dndarray.py [hunk 2]
```diff
@@ -730,6 +730,32 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
...
+                int(obj)
+                return True
+            except Exception:
+                pass
+            if hasattr(obj, "__len__") and len(obj) == 1:
+                return True
+            if hasattr(obj, "shape") and len(obj.shape) == 0:
+                return True
+            if hasattr(obj, "size"):
+                return obj.size == 1
```
Comment (ClaudiaComito) [inline heat/core/dndarray.py:748]: a torch tensor has attribute `size` as well, except...
```
>>> import torch
>>> a = torch.tensor(1)
>>> a.size()
torch.Size([])
```

**Review by ClaudiaComito — APPROVED** [review]
Great @ben-bou thanks, I'll merge it into the branch. The function probably belongs in the `sanitation` module, not in `dndarray`, but I can move it later.

**Pull request merged**